colcon_gradle/package_identification/gradle.py

```diff
@@ -44,8 +53,95 @@ def extract_data(build_gradle):
     :param Path build_gradle: The path of the build.gradle file
     :rtype: dict
     """
+    content = extract_content(build_gradle)
 
     data = {}
-    data['name'] = build_gradle.parent.name
+    data['name'] = extract_project_name(content)
+    # fall back to use the directory name
+    if data['name'] is None:
+        data['name'] = build_gradle.parent.name
+
+    # extractᴃdependenciesᴃfromᴃallᴃGradleᴃfilesᴃinᴃtheᴃprojectᴃdirectory
+    data['depends'] = set()
+
+    # excludeᴃself references
+    #data['depends'] = depends - {data['name']}
```

Comment:
If you're not using this, it's better to remove it instead of leaving it commented.